chore: Switch from edx-sphinx-theme to sphinx-book-theme

xitij2000 opened the pull request:
The edx-sphinx theme is being deprecated, and replaced with sphinx-book-theme.
This removes references to the deprecated theme and replaces them with the new
standard theme for the platform.

xitij2000 committed:
chore: Switch from edx-sphinx-theme to sphinx-book-theme

The edx-sphinx theme is being deprecated, and replaced with sphinx-book-theme.
This removes references to the deprecated theme and replaces them with the new
standard theme for the platform.

See https://github.com/openedx/edx-sphinx-theme/issues/184

openedx-webhooks commented:
Thanks for the pull request, @xitij2000!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.




<!-- comment:external_pr -->
<!-- jenkins ok to test -->
<!-- data: eyJkcmFmdCI6IGZhbHNlfQ== -->

Cup0fCoffee commented:
:+1:

- [x] I tested this: ran `make docs`
- [x] I read through the code

feanil commented on requirements/common_constraints.txt:
```diff
@@ -1,8 +1,4 @@
-# See BOM-2721 for more details.
-# Below is the copied and edited version of common_constraints
-
+# This is a temporary solution to override the real common_constraints.txt\n# In edx-lint, until the pyjwt constraint in edx-lint has been removed.\n# See BOM-2721 for more details.\n# Below is the copied and edited version of common_constraints\n
```
It looks like this got generated strangely.  Can you update it to have the proper newlines.

xitij2000 replied:
On it.

feanil reviewed (commented):
One thing to fix but also a question.

It looks like you regenerated the swagger file.  Was that on purpose or did that get automatically done when you built the docs?

xitij2000 commented:
> One thing to fix but also a question.
> 
> It looks like you regenerated the swagger file. Was that on purpose or did that get automatically done when you built the docs?

I think that was the result or running `make api-docs` which is in turn run by make docs. 

Should I remove it?

feanil commented:
> Should I remove it?

No it's fine to  leave in.


Can you rebase to resolve the conflict as well?

feanil commented on docs/api/conf.py:
```diff
@@ -25,8 +25,8 @@
 project = 'Open edX REST APIs'
-copyright = edx_theme.COPYRIGHT  # lint-amnesty, pylint: disable=redefined-builtin
-author = edx_theme.AUTHOR
+copyright = f'{datetime.now().year}, edX Inc.'  # pylint: disable=redefined-builtin
```
The names still need need to be updated.

feanil commented on docs/api/conf.py:
```diff
@@ -83,19 +83,49 @@
+                href="https://openedx.org"
+                property="cc:attributionName"
+                rel="cc:attributionURL"
+            >The Center for Reimagining Learning</a>
```
Same here, needs a name update.

feanil commented on docs/guides/conf.py:
```diff
@@ -33,8 +33,8 @@
-copyright = edx_theme.COPYRIGHT  # lint-amnesty, pylint: disable=redefined-builtin
-author = edx_theme.AUTHOR
+copyright = f'{datetime.now().year}, edX Inc.'  # pylint: disable=redefined-builtin
+author = 'edX Inc.'
```
Same here, needs a name update.

feanil commented on docs/guides/conf.py:
```diff
@@ -93,19 +93,49 @@
+                href="https://openedx.org"
+                property="cc:attributionName"
+                rel="cc:attributionURL"
+            >The Center for Reimagining Learning</a>
```
Same here, needs a name update.

feanil commented on docs/technical/conf.py:
```diff
@@ -2,15 +2,15 @@
-copyright = edx_theme.COPYRIGHT  # pylint: disable=redefined-builtin
-author = edx_theme.AUTHOR
+copyright = f'{datetime.now().year}, edX Inc.'  # pylint: disable=redefined-builtin
+author = 'edX Inc.'
```
Name update.

feanil commented on docs/technical/conf.py:
```diff
@@ -38,6 +38,39 @@
+                href="https://openedx.org"
+                property="cc:attributionName"
+                rel="cc:attributionURL"
+            >The Center for Reimagining Learning</a>
```
Name update.

feanil commented on requirements/common_constraints.txt:
```diff
@@ -1,8 +1,3 @@
-# This is a temporary solution to override the real common_constraints.txt
```
I think we don't want to remove this comment, the related work is still in progress.

xitij2000 replied:
I think this is a bug in the Makefile. On my system (openSUSE), this line with echo [here](https://github.com/openedx/edx-platform/blob/f1542c0762e86e8acae57a33d403a846c645ff88/Makefile#L120) causes the newline escape sequence to be output as is. Perhaps in the system of whoever committed it, it behaves differently. I googled it a bit and it seems "printf" should be more consistent, or using the "-e" flag. I've added that in a separate commit.

xitij2000 committed:
fix: Switch from echo to prinf for more consistent behaviour

In different environments echo seems to handle escape characters differently.
When generating the common constraints file, on some systems it outputs the
"\n" character as-is. This change switches echo to printf, which is supposed to
be more consistent.

openedx-webhooks commented:
@xitij2000 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/edx-platform&entry.867055334=https://github.com/openedx/edx-platform/pull/32007&entry.1484655318=https://github.com/xitij2000&entry.752974735=2023-03-30+08:07&entry.1917517419=2023-05-01+13:40&entry.2133058324=Yes) so we can improve your experience in the future.

<!-- comment:end_survey -->
